Objects/namespaceobject.c

```diff
@@ -43,10 +43,29 @@ namespace_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
 static int
 namespace_init(_PyNamespaceObject *ns, PyObject *args, PyObject *kwds)
 {
```

Comment:
(nothing to change here)

With this change it seems like `namespace_init()` is now almost identical `dict.__init__()` and `dict.update()`.  Could we take advantage of that?  Probably not since we have to verify string keys.

FWIW, I was going to make some suggestions (incl. exposing `dict_update_common()` in the internal C-API), until I remembered the string keys thing.  Maybe that function could have an option to check for string keys first (or a variant function that checks)?  However, it probably wouldn't be worth it without other use cases beside `SimpleNamespace`.